src/non-standalone/Dialog/ImageDialog.tsx

```diff
@@ -0,0 +1,269 @@
+		elem.multiple = false;
+		elem.addEventListener("change", () => {
+			const files = elem.files;
+			if (!files) return;
+			const file = files[0];
+			if (!file) return;
+			void processFile(file);
+		});
+		elem.click();
+	}, [processFile]);
+	const changePrimaryImage = useCallback(
+		(index) => {
+			const newImageArr = localuploadedImages.map((ele, ind) => {
+				if (index === ind) {
+					ele.setPrimary = !ele.setPrimary;
+				} else {
+					ele.setPrimary = false;
+				}
+				return ele;
+			});
+			setImages(newImageArr);
+			onPrimarySelected(newImageArr);
+		},
+		[localuploadedImages, onPrimarySelected]
+	);
+	const removeImageHandler = useCallback(
+		(index) => {
+			localuploadedImages.splice(index, 1);
+			setImages([...localuploadedImages]);
+			onUpdateImages(localuploadedImages);
+		},
+		[localuploadedImages, setImages, onUpdateImages]
+	);
+	const handleDrop = useCallback(
+		async (evt: React.DragEvent<HTMLDivElement>) => {
+			if (readOnly) return;
+
+			evt.preventDefault();
+
+			await processFile(evt.dataTransfer?.files[0]);
+		},
+		[readOnly, processFile]
+	);
+
+	const handleDragOver = useCallback(
+		(evt: React.DragEvent<HTMLDivElement>) => {
+			if (readOnly) return;
+
+			evt.preventDefault();
+		},
+		[readOnly]
+	);
+	return (
+		<Dialog
+			onClose={closeDialog}
+			maxWidth="lg"
+			fullWidth
+			disableBackdropClick
+			open={true}
+		>
+			<MuiDialogTitle disableTypography>
+				<IconButton
+					aria-label="close"
+					className={classes.closeButton}
+					onClick={closeDialog}
+				>
+					<Close />
+				</IconButton>
+			</MuiDialogTitle>
+			<DialogContent>
+				<Grid item>
+					<Grid container spacing={2}>
+						{localuploadedImages.map((image, i) => (
+							<Grid key={i}>
+								<Grid item xs className={classes.layout}>
+									{!image.setPrimary && (
+										<Cancel
+											onClick={() => removeImageHandler(i)}
+											className={classes.closeIcon}
+										/>
+									)}
+									<img
+										src={image.src}
+										alt="img"
+										className={classes.addPhotoContainer}
+									/>
+								</Grid>
+								<Grid container alignItems="center" style={{ margin: "30px" }}>
+									<Star
+										color={image.setPrimary ? "primary" : "disabled"}
+										onClick={() => changePrimaryImage(i)}
+									/>
+									{ccI18n.t(
+										"standalone.image-box-control.dialog.primary-button"
+									)}
+								</Grid>
+							</Grid>
+						))}
+						{!readOnly && (
+							<Grid className={classes.layoutAdd}>
+								<Grid item xs onDrop={handleDrop} onDragOver={handleDragOver}>
+									<div className={classes.addPhotoContainer}>
+										<AddAPhoto
+											style={{
+												height: "50px",
+												width: "50px",
+												cursor: "pointer",
+											}}
+											className={classes.addPhoto}
+											onClick={handleUpload}
+										/>
+									</div>
+									<input
+										type={"text"}
+										ref={changeRef}
```

Comment:
Add the name attribute and make it settable via props, otherwise this whole event stuff is useless